src/sentry/event_manager.py

```diff
@@ -600,9 +600,21 @@ def save(self, project, raw=False):
         if release:
             group_kwargs['first_release'] = release
 
-        group, is_new, is_regression, is_sample = self._save_aggregate(
-            event=event, hashes=hashes, release=release, **group_kwargs
-        )
+        try:
+            group, is_new, is_regression, is_sample = self._save_aggregate(
+                event=event, hashes=hashes, release=release, **group_kwargs
+            )
+        except HashDiscarded:
+            event_discarded.send_robust(
+                project=project,
+                sender=EventManager,
+            )
+            raise
```

Comment:
i'm just re-raising here because some stuff in the `save_event` task still relies on this error. I will most likely move that here too in a future pr and then can get rid of this